src/psm/src/ir_solver.cpp

```diff
@@ -271,12 +272,17 @@ bool IRSolver::AddC4Bump()
   if (m_C4Bumps.size() == 0) {
     m_logger->error(utl::PSM, 14, "Number of voltage sources cannot be 0.");
   }
-  m_logger->info(utl::PSM, 64, "Number of voltage sources = {}.", m_C4Bumps.size());
-  for (size_t it = 0; it < m_C4Nodes.size(); ++it) {
-    NodeIdx node_loc      = m_C4Nodes[it].first;
-    double  voltage_value = m_C4Nodes[it].second;
-    m_Gmat->AddC4Bump(node_loc, it);  // add the 0th bump
-    m_J.push_back(voltage_value);     // push back first vdd
+  m_logger->info(utl::PSM, 64, "Number of voltage sources = {}.", 
+                 m_C4Bumps.size(),m_C4Nodes.size());
+  std::map<NodeIdx, double>::iterator c4_node_it;
+  size_t it = 0;
+  for (c4_node_it = m_C4Nodes.begin(); c4_node_it != m_C4Nodes.end();
+       c4_node_it++) {
```

Comment:
More compact, modern style:

for (auto [node_loc, voltage_value] : m_C4Nodes) { ... }